src/amqp10/messageAnnotations.ts

```diff
@@ -0,0 +1,20 @@
+import { MessageAnnotations } from "../producer"
+import { DataReader } from "../responses/raw_response"
+import { range } from "../util"
+import { readUTF8String, decodeFormatCode } from "../response_decoder"
+
+export class Annotations {
+  public static parse(dataReader: DataReader, elementsLength: number): MessageAnnotations {
+    const numEntries = elementsLength / 2
+
+    return range(numEntries).reduce((acc: MessageAnnotations, _) => {
+      const propertyKey = readUTF8String(dataReader)
+      const nextByteType = dataReader.readUInt8()
+      dataReader.rewind(1)
+      const propertyValue = decodeFormatCode(dataReader, nextByteType, true)
+      if (!propertyValue) throw new Error(`invalid nextByteType %#02x: ${nextByteType}`)
+      acc[propertyKey] = propertyValue as string | number
```

Comment:
Are you sure about this `as` ?

Reply:
I don't know how to do it otherwise, decodeFormatCode has multiple return types